core/http/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/transaction/Transaction.java

```diff
@@ -113,13 +117,12 @@ UUID getID() {
 	void begin(IsolationLevel level)
 		throws InterruptedException, ExecutionException
 	{
-		Future<Boolean> result;
-		synchronized (futures) {
-			result = executor.submit(() -> {
-				txnConnection.begin(level);
-				return true;
-			});
+		Future<Boolean> result = executor.submit(() -> {
+			txnConnection.begin(level);
+			return true;
+		});
 
+		synchronized (futures) {
```

Comment:
This was done because threads need concurrent access to `futures` to determine if operations are currently active. See `hasActiveOperations`. The order in which the Future objects get added to the list is not really relevant, but this is done in a synchronized manner because `ArrayList` is not thread-safe. 

I think that this should  indeed be reverted.